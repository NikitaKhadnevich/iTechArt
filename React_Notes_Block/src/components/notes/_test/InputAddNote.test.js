/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import InputAddNote from '../inputAddNote/InputAddNote';

configure({ adapter: new Adapter() });

describe('InputAddNote', () => {
  const valuesData = { title: 'some title', description: 'some description' };
  const addNoteProps = {
    formik: {
      values: { title: 'some title', description: 'some description' },
      errors: {
        title: 'are title Error',
        description: 'are description Error',
      },
      touched: {
        title: 'are title Error',
        description: 'are descrbtion Error',
      },
      handleChange: jest.fn(),
      isValid: false,
      setFieldTouched: jest.fn(),
      handleBlur: jest.fn(),
      handleSubmit: jest.fn(),
    },
  };

  const addNotePropsDone = {
    formik: {
      values: { title: '', description: '' },
      errors: {
        title: '',
        description: '',
      },
      touched: {
        title: '',
        description: '',
      },
      handleChange: jest.fn(),
      isValid: false,
      setFieldTouched: jest.fn(),
      handleBlur: jest.fn(),
      handleSubmit: jest.fn(),
    },
  };

  const {
    formik: {
      handleSubmit,
      errors: { title, description },
    },
  } = addNoteProps;

  test('drink returns La Croix', () => {
    const beverage = { name: 'La Croix' };
    const drink = jest.fn((beverage) => beverage.name);

    drink(beverage);

    expect(drink).toHaveReturnedWith('La Croix');
  });

  it('Is it InputAddNote Exist', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    expect(addNote).toMatchSnapshot();
    expect(addNote).toHaveLength(1);
  });
  it('InputAddNote Has only 2 fields', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    expect(addNote.find('ForwardRef(TextField)')).toHaveLength(2);
  });
  it('InputAddNote Has Title field', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    expect(addNote.find('ForwardRef(TextField)').first().props().id).toEqual(
      'title'
    );
  });
  it('InputAddNote Has Description field', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    const notesFields = addNote.findWhere(
      (n) => n.type('ForwardRef(TextField)') && n.props().id === 'description'
    );
    expect(notesFields).toHaveLength(1);
  });
  it('Each fields are Required', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    const notesFields = addNote.findWhere(
      (n) => n.type('ForwardRef(TextField)') && n.props().required === true
    );
    expect(notesFields).toHaveLength(2);
  });

  it('There are No errors inputs Title', () => {
    const addNote = shallow(<InputAddNote {...addNotePropsDone} />);
    const {
      formik: {
        errors: { title, description },
      },
    } = addNotePropsDone;
    const notesFields = addNote.findWhere(
      (n) => n.type('ForwardRef(TextField)') && n.props().helperText === title
    );
    expect(notesFields).toHaveLength(2);
  });
  it('There are No errors inputs Description', () => {
    const addNote = shallow(<InputAddNote {...addNotePropsDone} />);
    const {
      formik: {
        errors: { title, description },
      },
    } = addNotePropsDone;
    const notesFields = addNote.findWhere(
      (n) =>
        n.type('ForwardRef(TextField)') && n.props().helperText === description
    );
    expect(notesFields).toHaveLength(2);
  });

  it('There are error In Title', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    const notesFields = addNote.findWhere(
      (n) => n.type('ForwardRef(TextField)') && n.props().helperText === title
    );
    expect(notesFields).toHaveLength(1);
  });
  it('There are error In Description', () => {
    const addNote = shallow(<InputAddNote {...addNoteProps} />);
    const notesFields = addNote.findWhere(
      (n) =>
        n.type('ForwardRef(TextField)') && n.props().helperText === description
    );
    expect(notesFields).toHaveLength(1);
  });
  it('Test Add Note Event', () => {
    const actionForm = shallow(<InputAddNote {...addNoteProps} />);
    const {
      formik: { handleSubmit, values },
    } = addNoteProps;
    actionForm.find('Styled(ForwardRef(Box))').simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
