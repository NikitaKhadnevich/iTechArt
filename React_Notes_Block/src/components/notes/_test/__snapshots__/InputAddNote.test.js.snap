// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAddNote Is it InputAddNote Exist 1`] = `
<Styled(ForwardRef(Box))
  component="form"
  onSubmit={[MockFunction]}
>
  <Styled(ForwardRef(Grid))
    container={true}
    item={true}
    md={9}
    sm={10}
    spacing={1}
    xs={12}
  >
    <ForwardRef(Grid)
      alignItems="center"
      item={true}
      md={3}
      sm={4}
      xs={12}
    >
      <ForwardRef(TextField)
        error={true}
        helperText="are title Error"
        id="title"
        label="Title"
        onBlur={[MockFunction]}
        onChange={[Function]}
        required={true}
        size="small"
        sx={
          Object {
            "width": "100%",
          }
        }
        value="some title"
        variant="filled"
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      md={5}
      sm={8}
      xs={12}
    >
      <ForwardRef(TextField)
        error={true}
        helperText="are description Error"
        id="description"
        label="Describe"
        onBlur={[MockFunction]}
        onChange={[Function]}
        required={true}
        size="small"
        sx={
          Object {
            "width": "100%",
          }
        }
        value="some description"
        variant="filled"
      />
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      md={2}
      sm={4}
      xs={12}
    >
      <SortNotes />
    </ForwardRef(Grid)>
    <Styled(ForwardRef(Grid))
      item={true}
      md={2}
      sm={8}
      xs={12}
    >
      <Styled(WithStyles(ForwardRef(Button)))
        type="submit"
      >
        Add Note
      </Styled(WithStyles(ForwardRef(Button)))>
    </Styled(ForwardRef(Grid))>
  </Styled(ForwardRef(Grid))>
</Styled(ForwardRef(Box))>
`;
